var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -483,30 +470,26 @@ def cmake_args(self):
             ])
 
         # ScaLAPACK
+        options.append(self.define_from_variant(
+            'DEAL_II_WITH_SCALAPACK', 'scalapack'
+        ))
         if '+scalapack' in spec:
             scalapack = spec['scalapack'].libs
             options.extend([
                 '-DSCALAPACK_FOUND=true',
                 '-DSCALAPACK_INCLUDE_DIRS=%s' % (
                     spec['scalapack'].prefix.include),
-                '-DSCALAPACK_LIBRARIES=%s' % scalapack.joined(';'),
-                '-DDEAL_II_WITH_SCALAPACK=ON'
-            ])
-        else:
-            options.extend([
-                '-DDEAL_II_WITH_SCALAPACK=OFF'
+                '-DSCALAPACK_LIBRARIES=%s' % scalapack.joined(';')
```

Comment:
The static `CMakePackage.define(key, value)` method formats the `value` according to the type: so if you do
```suggestion
                self.define('SCALAPACK_LIBRARIES', scalapack)
```
it'll do the list-to-cmake conversions for you. Likewise with boolean arguments. There are a couple of other places you could to this if you want.

Reply:
Thanks for the suggestion! It's great that the CMake syntax is now taken care of by the package. There were a lot of changes that came from employing this approach, so I'm going to need a little time to check that there are no negative consequences from using it.